src/pratt.rs

```diff
@@ -213,31 +235,30 @@ impl<'a, I, O, E, Atom, InfixOps, InfixOpsOut, PrefixOps, PrefixOpsOut>
     }
 }
 
-impl<'a, I, O, E, Atom, InfixOps, InfixOpsOut, PostfixOps, PostfixOpsOut>
-    Pratt<I, O, E, Atom, InfixPostfix<InfixOps, InfixOpsOut, PostfixOps, PostfixOpsOut>>
+impl<'a, I, O, E, Atom, InfixParser, InfixOperator, PostfixParser, PostfixOperator>
+    Pratt<I, O, E, Atom, InfixPostfix<InfixParser, InfixOperator, PostfixParser, PostfixOperator>>
 {
     /// Extend a `Pratt` parser by setting prefix operators
-    pub fn with_prefix_ops<PrefixOps, PrefixOpsOut>(
+    pub fn with_prefix_ops<PrefixParser, PrefixOperator>(
         self,
-        prefix_ops: PrefixOps,
+        prefix_ops: PrefixParser,
     ) -> Pratt<
         I,
         O,
         E,
         Atom,
         InfixPrefixPostfix<
-            InfixOps,
-            InfixOpsOut,
-            PrefixOps,
-            PrefixOpsOut,
-            PostfixOps,
-            PostfixOpsOut,
+            InfixParser,
+            InfixOperator,
+            PrefixParser,
+            PrefixOperator,
+            PostfixParser,
+            PostfixOperator,
         >,
     >
     where
         I: Input<'a>,
         E: ParserExtra<'a, I>,
-        PrefixOps: Parser<'a, I, PrefixOpsOut, E>,
```

Comment:
Any particular reason this trait bound has to disappear? It was originally added with the idea that it makes no sense to construct a Parser which cannot parse.